File: php/features.php
<?php
declare(strict_types=1);

// Argentinadatos SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ArgentinadatosFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ArgentinadatosBaseFeature();
            case "test":
                return new ArgentinadatosTestFeature();
            default:
                return new ArgentinadatosBaseFeature();
        }
    }
}

File: py/features.py
# Argentinadatos SDK feature factory

from feature.base_feature import ArgentinadatosBaseFeature
from feature.test_feature import ArgentinadatosTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ArgentinadatosBaseFeature(),
        "test": lambda: ArgentinadatosTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
